src/main/java/com/arpnetworking/tsdcore/model/CompositeUnit.java

```diff
@@ -0,0 +1,338 @@
+    }
+
+    public void setType(final Type value) {
+        _type = value;
+    }
+
+    public void setScale(final Scale value) {
+        _scale = value;
+    }
+
+    /**
+     * Composite unit data type.
+     */
+    public enum Type {
+        /*****************************************************************
+         * Time
+         */
+        /**
+         * Second.
+         */
+        SECOND(Category.TIME, 1),
+        /**
+         * Minute, 60 seconds.
+         */
+        MINUTE(Category.TIME, 60),
+        /**
+         * Hour, 60 minutes.
+         */
+        HOUR(Category.TIME, 60 * 60),
+        /**
+         * Day, 24 hours.
+         */
+        DAY(Category.TIME, 60 * 60 * 24),
+        /**
+         * Week, 7 days.
+         */
+        WEEK(Category.TIME, 60 * 60 * 24 * 7),
+        /*****************************************************************
+         * Data Size
+         */
+        /**
+         * Bit.
+         */
+        BIT(Category.DATA_SIZE, 1),
+        /**
+         * Byte, 8 bits.
+         */
+        BYTE(Category.DATA_SIZE, 8),
+        /*****************************************************************
+         * Rotation
+         */
+        /**
+         * Rotation.
+         */
+        ROTATION(Category.ROTATION, 1),
+        /**
+         * Degree, 360 = 1 Rotation.
+         */
+        DEGREE(Category.ROTATION, 360),
+        /**
+         * Radian, 2PI = 1 Rotation.
+         */
+        RADIAN(Category.ROTATION, 2 * Math.PI),
+        /*****************************************************************
+         * Temperature
+         */
+        /**
+         * Celsius.
+         */
+        CELSIUS(Category.TEMPERATURE, 1),
+        /**
+         * Fahrenheit.
+         */
+        FAHRENHEIT(Category.TEMPERATURE, 1),
+        /**
+         * Kelvin.
+         */
+        KELVIN(Category.TEMPERATURE, 1);
+
+        Type(final Category category, final double scaler) {
+            _category = category;
+            _scaler = scaler;
+        }
+
+        private Category _category;
+        private double _scaler;
+    }
+
+    /**
+     * Scalar values.
+     */
+    public enum Scale {
+        /**
+         * 10^-24.
+         */
+        YOCTO(10 ^ -24),
+        /**
+         * 10^-21.
+         */
+        ZEPTO(10 ^ -21),
+        /**
+         * 10^-18.
+         */
+        ATTO(10 ^ -18),
+        /**
+         * 10^-15.
+         */
+        FEMTO(10 ^ -15),
+        /**
+         * 10^-12.
+         */
+        PICO(10 ^ -12),
+        /**
+         * 10^-9.
+         */
+        NANO(10 ^ -9),
+        /**
+         * 10^-6.
+         */
+        MICRO(10 ^ -6),
+        /**
+         * 10^-3.
+         */
+        MILLI(10 ^ -3),
+        /**
+         * 10^-2.
+         */
+        CENTI(10 ^ -2),
+        /**
+         * 10^-1.
+         */
+        DECI(10 ^ -1),
+        /**
+         * 1.
+         */
+        ONE(1),
```

Comment:
Since it's unclear whether this is an internal model or a format model, my comment may or
May not be correct: but ONE afaik is not a valid value in the scheme. Right?